i18n/zh/docusaurus-plugin-content-docs/current/architecture/edge/devicetwin.md

```diff
@@ -2,353 +2,327 @@
-3. For each message the action message is read and the corresponding function is called
-4. Receive heartbeat from the heartbeat channel and send a heartbeat to the controller
+该模块执行的主要功能是：
 
-The following are the action callbacks which can be performed by the membership module:
+1. 初始化动作回调映射，是一个 map[string] Callback，包含可以执行的回调函数
+2. 接收发送到 Membership 模块的消息
+3. 对于每条消息，都会读取动作消息并调用相应的函数
+4. 从心跳通道接收心跳，并向控制器发送心跳
+
+以下是 Membership 模块可以执行的操作回调：
 
    - dealMembershipGet
    - dealMembershipUpdated
    - dealMembershipDetail
 
-**dealMembershipGet**: dealMembershipGet() gets the information  about the devices associated with the particular edge node from the cache.
-   - The eventbus first receives a message on its subscribed topic (membership-get topic).
-   - This message arrives at the  devicetwin controller, which further sends the message to membership module.
-   - The membership module gets the devices associated with the edge node from the cache (context) and sends the information to the communication module.It also handles errors that may arise while performing the  aforementioned process and sends the error to the communication module instead of device details.
-   - The communication module sends the information to the  eventbus component which further publishes the result on the specified MQTT topic (get membership result topic).
+**dealMembershipGet**: dealMembershipGet() 从缓存中获取与特定边缘节点关联的设备的信息。
+   - eventbus 首先在其订阅的主题（membership-get）上接收一条消息。
+   - 此消息到达 DeviceTwin Controller，由它进一步将消息发送到 Membership 模块。
+   - Membership 从缓存中获取与边缘节点关联的设备，并将信息发送到通信模块。它还处理在执行上述过程时可能出现的错误，并将错误发送到通信模块而不是设备详细信息。
+   - 通信模块将信息发送给 eventbus 组件，eventbus 组件进一步在指定的 MQTT topic上发布结果。
 
   ![Membership Get()](/img/devicetwin/membership-get.png)
 
 
-**dealMembershipUpdated**: dealMembershipUpdated() updates the membership details of the node. It adds the devices, that were newly added, to the edge group and removes the devices, that were removed, from the edge group and updates device details, if they have been altered or updated.
-   - The edgehub module receives the membership update message from the cloud and forwards the message to devicetwin controller which further forwards it to the membership module.
-   - The membership  module adds devices that are newly added, removes devices that have been recently deleted and also updates the devices that were already existing in the database as well as in the cache.
-   - After updating the details of the devices a  message is sent to the communication module of the device twin, which sends the message to eventbus module to be published on the given MQTT topic.
+**dealMembershipUpdated**: dealMembershipUpdated() 函数用于更新节点的成员资格详情。它将新添加的设备添加到边缘组中，将已移除的设备从边缘组中删除，并在设备详情发生变更时进行更新。
+   - Edgehub 模块从云端接收成员资格更新消息，并将该消息转发到 DeviceTwin Controller，后者进一步将其转发到 Membership 模块。
+   - Membership 模块添加新添加的设备、删除最近删除的设备以及更新数据库和缓存中已存在的设备。
+   - 更新设备的详细信息后，一条消息将发送到设备孪生的通信模块，该模块将消息发送到要在给定 MQTT 主题上发布的 eventbus 模块。
```

Comment:
```suggestion
   - 更新设备的详细信息后，一条消息将发送到DeviceTwin的Communication模块，该模块将消息发送到要在给定 MQTT 主题上发布的 eventbus 模块。
```
The module name in the document can be directly referred to by its English name, and does not need to be forcibly translated into Chinese.